### PR TITLE
fix: update mark price only once per block

### LINKED ISSUE #1354: Mark-to-market only at timestamp change...
... as a amplified spam mitigation, see https://github.com/vegaprotocol/specs/issues/1326.

### PR DESCRIPTION
**Pull request by davidsiska-vega**
Closes https://github.com/vegaprotocol/specs/issues/1354

### CODE REVIEW

**Comment by witgaw** [comment]
Could we not have it update once `n` blocks where `n` is either hardcoded or a network parameter?

Slightly more work with the counter etc., but more flexibility for the future.

Could also allow `n == 0` which would preserve current behaviour, again, more work, but perhaps a saving of effort overall as we wouldn't have to update the existing tests.

**Comment by davidsiska-vega** [comment]
> Could we not have it update once `n` blocks where `n` is either hardcoded or a network parameter?
> 
> Slightly more work with the counter etc., but more flexibility for the future.
> 
> Could also allow `n == 0` which would preserve current behaviour, again, more work, but perhaps a saving of effort overall as we wouldn't have to update the existing tests.

I broadly agree; though do we want to tie it to blocks or would we rather have passage of time (e.g. at least 5s elapsed)? I agree to include `0s` which would preserve current behaviour.

**Comment by witgaw** [comment]
> > Could we not have it update once `n` blocks where `n` is either hardcoded or a network parameter?
> > Slightly more work with the counter etc., but more flexibility for the future.
> > Could also allow `n == 0` which would preserve current behaviour, again, more work, but perhaps a saving of effort overall as we wouldn't have to update the existing tests.
> 
> I broadly agree; though do we want to tie it to blocks or would we rather have passage of time (e.g. at least 5s elapsed)? I agree to include `0s` which would preserve current behaviour.

Yeah, that sounds even better.

File: protocol/0009-MRKP-mark_price.md [hunk 1]
```diff
@@ -2,16 +2,20 @@ Feature name: Mark Price
 # Summary
 A *Mark Price* is a concept derived from traditional markets.  It is a calculated value for the 'current market price' on a market.
 
+Introduce a network parameter `network.markPriceUpdateMaximumFrequency` with minimum allowable value of `0s` maximum allowable value of `1h` and a default of `5s`.
```
Comment (witgaw) [inline protocol/0009-MRKP-mark_price.md:5]: Isn't it just `network.markPriceUpdateFrequency`? Or are you trying to convey that exact timings are not guaranteed?

**Review by witgaw — APPROVED** [review]
LGTM, just a small comment/question:

**Pull request merged**